ledger-core/virtual/execute/execute.go

```diff
@@ -1001,6 +1002,54 @@ func (s *SMExecute) stepFinishRequest(ctx smachine.ExecutionContext) smachine.St
 	return ctx.Stop()
 }
 
+func (s *SMExecute) stepSendExistingCallResult(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	var (
+		msg   *payload.VCallResult
+		found bool
+	)
+
+	action := func(state *object.SharedState) {
+		results := state.KnownRequests.GetResults()
+
+		summary, ok := results[s.execution.Outgoing]
```

Comment:
why don't you get this result from shared data during deduplication step?

Reply:
You prefer to extract results to some field and reused it in the send step? It was not clean code, what problem with my approach?

Reply:
I believe that you don't need any step if you organize code differently. If you move `s.messageSender.PrepareAsync` and a bit of code around into a method "sendResultMessage" or something like that then you will have "one line call" that would allow you to send result right from stepDeduplicate without additional step at all.

Reply:
I will check, but it may more affected code

Reply:
It will work only if we will pass some parameter in stepSendMessage for a jump to the next step. In normal flow after stepSendMessage we jump to stepFinishRequest, inflow deduplication, we will stop. This mean we should have flag or enum in SMExecute fields. Do you think it it will be better?

Reply:
Not step, but method. Just a function that abstracts repeated code. Our SMs shouldn't be just steps and migrations. They may have methods that help us organize code.

Reply:
I can show a diff if you like

Reply:
It was in my question before)

Reply:
Fixed